tutorials/preprocessing/plot_40_artifact_correction_ica.py

```diff
@@ -1,307 +1,549 @@
+#     Sometimes the right tool is no tool at all — if the artifacts are small
+#     enough you may not even need to repair them to get good analysis results.
+#     See :ref:`tut-artifact-overview` for guidance on detecting and
+#     visualizing various types of artifact.
+#
+#
+# What is ICA?
+# ^^^^^^^^^^^^
+#
+# Independent components analysis (ICA) is a technique for estimating
+# independent source signals from a set of recordings in which the source
+# signals were mixed together in unknown ratios. A common example of this is
+# the problem of `blind source separation`_: with 3 musical instruments playing
+# in the same room, and 3 microphones recording the performance (each picking
+# up all 3 instruments, but at varying levels), can you somehow "unmix" the
+# signals recorded by the 3 microphones so that you end up with a separate
+# "recording" isolating the sound of each instrument?
+#
+# It is not hard to see how this analogy applies to EEG/MEG analysis: there are
+# many "microphones" (sensor channels) simultaneously recording many
+# "instruments" (blinks, heartbeats, activity in different areas of the brain,
+# muscular activity from jaw clenching or swallowing, etc). As long as these
+# various source signals are `statistically independent`_ and non-gaussian, it
+# is usually possible to separate the sources using ICA, and then re-construct
+# the sensor signals after excluding the sources that are unwanted.
+#
+#
+# ICA in MNE-Python
+# ~~~~~~~~~~~~~~~~~
+#
+# MNE-Python implements three different ICA algorithms: ``fastica`` (the
+# default), ``picard``, and ``infomax``. FastICA and Infomax are both in fairly
+# widespread use; Picard is a newer (2017) algorithm that is expected to
+# converge faster than FastICA and Infomax, and is more robust than other
+# algorithms in cases where the sources are not completely independent, which
+# typically happens with real EEG/MEG data. See [1]_ for more information.
+#
+# The ICA interface in MNE-Python mirrors the standard interface in
+# `scikit-learn`_: some general parameters are specified when creating an
+# :class:`~mne.preprocessing.ICA` object, then the
+# :class:`~mne.preprocessing.ICA` object is fit to the data using its
+# :meth:`~mne.preprocessing.ICA.fit` method. The results of the fitting are
+# added to the :class:`~mne.preprocessing.ICA` object as attributes that end in
+# an underscore (``_``), such as ``ica.mixing_matrix_`` and
+# ``ica.unmixing_matrix_``. After fitting, the ICA component(s) that you want
+# to remove must be chosen, and the ICA fit must then be applied to the
+# :class:`~mne.io.Raw` or :class:`~mne.Epochs` object using the
+# :class:`~mne.preprocessing.ICA` object's :meth:`~mne.preprocessing.ICA.apply`
+# method.
+#
+# As is typically done with ICA, the data are first
+# scaled to unit variance and whitened using principal components analysis
+# (PCA) before performing the ICA decomposition. You can impose a
```

Comment:
is this to speed up computation or because you assume that the noise in the model is white with identity covariance?

Reply:
... oh wait, the [rendered version](https://14439-1301584-gh.circle-artifacts.com/0/dev/auto_tutorials/preprocessing/plot_40_artifact_correction_ica.html) does not match the text. I guess this has been fixed.